ee/app/livechat-enterprise/server/hooks/onLoadConfigApi.js

```diff
@@ -1,35 +1,16 @@
 import { callbacks } from '../../../../../app/callbacks';
-import { settings } from '../../../../../app/settings';
-import { LivechatInquiry } from '../../../../../app/models/server';
-import { normalizeQueueInfo } from '../lib/Helper';
+import { getLivechatQueueInfo, getLivechatCustomFields } from '../lib/Helper';
 
-callbacks.add('livechat.onLoadConfigApi', async (room) => {
-	if (!room) {
-		return null;
-	}
+callbacks.add('livechat.onLoadConfigApi', async (options = {}) => {
+	const { room } = options;
 
-	if (!settings.get('Livechat_waiting_queue')) {
-		return null;
-	}
+	const queueInfo = await getLivechatQueueInfo(room);
+	const customFields = getLivechatCustomFields();
 
-	const { _id: rid } = room;
-	const inquiry = LivechatInquiry.findOneByRoomId(rid);
-	if (!inquiry) {
-		return null;
-	}
+	const config = {
+		...queueInfo && { queueInfo },
+		...customFields && { customFields },
+	};
 
-	const { _id, status } = inquiry;
-	if (status !== 'queued') {
-		return null;
-	}
-
-	const [inq] = await LivechatInquiry.getCurrentSortedQueueAsync({ _id });
-
-	let queueInfo;
-	if (inq) {
-		const { position, department } = inq;
-		queueInfo = await normalizeQueueInfo({ position, department });
-	}
-
-	return { queueInfo };
+	return Object.assign({ config }, { ...options });
```

Comment:
you are creating a new object to assign a another new object... there are many ways to do that... I will suggest some of them 
```javascript
     return { config , ...options };
     return Object.assign({ config }, options);

or 
return  { 
   config: { 
      ...queueInfo && { queueInfo },
      ...customFields && { customFields },
   },
   ...options,
}